dll/win32/shell32/shlfileop.cpp

```diff
@@ -1630,6 +1639,8 @@ static void move_dir_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, LPCWST
 
     destroy_file_list(&flFromNew);
     destroy_file_list(&flToNew);
+
+    Win32RemoveDirectoryW(feFrom->szFullPath);
```

Comment:
I would not remove the original directory in case moving the files didn't succeed (aka I would check for move_files to succeed completely in moving **all** the files into the destination).

Reply:
Checked by `PathIsDirectoryEmptyW`. OK.